api/graphql/query/CalendarEvent/invites.js

```diff
@@ -7,7 +7,9 @@ const invites = async (parent, _args, ctx) => {
     where: { event_id: parent.id },
   };
 
-  if (
+  if (parent?.is_public && !ctx.viewer) {
```

Comment:
This is not the solution. The policy should not lie outside the AuthPolicy. 

In the where clause, include the condition checking for user_id only if the viewer is present, otherwise don't have that in where clause.